### ISSUE TITLE
Understanding Sample Homography

### ISSUE THREAD

**Issue by heiko134470** [rpautrat/SuperPoint#172]
![image](https://user-images.githubusercontent.com/58684478/97303678-a9a09800-1895-11eb-9c33-17de910b71cf.png)

Hi, firstly I would like to thank you for your work on this repository.

I would like to understand how to sample homography, especially through perspective perturbations. 
What does mean h_displacement_left, h_displacement_right, perspective_displacement?
Could you provide the detailed geometric theory?

**Comment by rpautrat** [comment]
Hi,

To generate a random homography, we start from four points on the corners of a square and distort them with several transformations. h_displacement_left will translate the two points on the left of the square to the left, same for h_displacement_right to the right for the two corners on the right, and perspective_displacement is the amount by which the top corners will translated to the top and the bottom corners to the bottom.
In other words, each corner is translated to the outside of the corner, and if you use different values for h_displacement_left, h_displacement_right and perspective_displacement, you will get a perspective effect.
